### PR TITLE
syncer: check block-allow list before online DDL components

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

close https://github.com/pingcap/dm/issues/1776

### What is changed and how it works?

move block-allow list check and binlog filter check before online DDL components

### Check List 

Tests 

 - Integration test

Code changes

 - Has interface methods change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

File: syncer/ddl.go [hunk 1]
```diff
@@ -113,26 +115,55 @@ func (s *Syncer) parseDDLSQL(sql string, p *parser.Parser, schema string) (resul
...
+		}
+
+		// get real tableNames before apply block-allow list
+		if s.onlineDDL != nil {
+			for _, names := range tableNames {
+				names.Name = s.onlineDDL.RealName(names.Name)
+			}
+		}
+
+		ignore, err2 := s.skipQuery(tableNames, stmt2, sql)
```
Comment (Ehco1996) [inline syncer/ddl.go:153]: if we skipQuery here, do we need skipQuery in [parseDDLSQL](https://github.com/pingcap/dm/blob/d2a1e30cb2588066d86e6830979fa03e1df8f8af/syncer/ddl.go#L36)

but the `stmt` here is from parseDDLSQL and some users' sql will be rewritten and I'm not sure how to refine the skipQuery logic here.

Comment (Ehco1996) [inline syncer/ddl.go:153]:
```suggestion
		skipped, err2 := s.skipQuery(tableNames, stmt2, sql)
```

**Review by Ehco1996 — COMMENTED** [review]
rest LGTM

File: syncer/ddl.go (hunk 1 above)
Comment (lance6716) [reply syncer/ddl.go]: https://github.com/pingcap/dm/pull/1775/commits/dc35699ad8e8f71536f17cf54c0008e54b77b0c7

**Review by Ehco1996 — APPROVED** [review]

File: syncer/ddl_test.go [hunk 2]
```diff
@@ -18,8 +18,11 @@ import (
 	"errors"
 	"fmt"
 
+	"go.uber.org/zap"
```
Comment (lichunzhu) [inline syncer/ddl_test.go:21]: group imports

**Review by lichunzhu — APPROVED** [review]
LGTM

**Review by glorv — APPROVED** [review]

**Pull request merged**